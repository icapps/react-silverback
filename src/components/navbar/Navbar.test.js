import React from 'react';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import '@testing-library/jest-dom/extend-expect';

import Navbar from './NavBar';

const history = createMemoryHistory({ initialEntries: ['/'] });

describe('Navbar Component', () => {
  it('should render a Navbar component', () => {
    const linkPath1 = 'test1';
    const linkPath2 = 'test2';
    const buildNr = 'V1';
    const versionNr = '1.2.3';

    const { queryByText } = render(
      <Router history={history}>
        <Navbar
          links={[
            { name: linkPath1, path: '/test1' },
            { name: linkPath2, path: '/test2' },
          ]}
          toggleNavigation={() => {}}
          isNavigationShown={true}
          build={buildNr}
          version={versionNr}
        />
      </Router>,
    );
    expect(queryByText(linkPath1)).toBeInTheDocument();
    expect(queryByText(linkPath2)).toBeInTheDocument();
    expect(queryByText(`Version: ${versionNr}`)).toBeInTheDocument();
    expect(queryByText(`Build: ${buildNr}`)).toBeInTheDocument();
  });
});
