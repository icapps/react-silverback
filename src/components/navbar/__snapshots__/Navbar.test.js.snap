// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component should render a Navbar component 1`] = `ShallowWrapper {}`;

exports[`Navbar Component should render a Navbar component with navigation shown 1`] = `ShallowWrapper {}`;

exports[`Navbar Component should render a hidden Navbar component 1`] = `ShallowWrapper {}`;
